lib/logstash/inputs/kinesis.rb

```diff
@@ -49,8 +51,15 @@ class LogStash::Inputs::Kinesis < LogStash::Inputs::Base
   # to enable the cloudwatch integration in the Kinesis Client Library.
   config :metrics, :validate => [nil, "cloudwatch"], :default => nil
 
-  def initialize(params = {}, kcl_class = KCL::Worker)
-    @kcl_class = kcl_class
+  #what a nasty hack to use the overloaded method :_(
+  java_import KCL::Worker::Builder
+  class Builder
+    #I don't know how to make it work using the class directly
+    java_alias :v2RecordProcessorFactory, :recordProcessorFactory, [Java::JavaClass.for_name(KCL_PROCESSOR_FACTORY_CLASS)]
```

Comment:
![no idea](https://stubbornthoughts.files.wordpress.com/2013/02/ihavenoideawhatimdoingthread_06af9676f192d84f17cf5a5816dc7523.jpg)

Jokes aside, I've used the `java_alias` because, as I learned yesterday you can overload a method in Ruby (using the same method name with diferent signature), and the KCL Builder now exposes two methods with the name `recordProcessorFactory`: [one](https://github.com/awslabs/amazon-kinesis-client/blob/master/src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java#L687) with the v1 recordProcessorFactory interface (wrapping it with an v1-to-v2 adaptor) and the [good one](https://github.com/awslabs/amazon-kinesis-client/blob/master/src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java#L701) which uses the v2 recordProcessorFactory.

So, in order to preserve the ability to test the worker and to use the v2 API I had to use a java_alias, to map the v2 API method to a new name. 

As a disclaimer I have to say that I don't know if this is the way to do this, but I didn't find any other way to hack it.